dask/dataframe/optimize.py

```diff
@@ -26,5 +31,9 @@ def optimize(dsk, keys, **kwargs):
     except ImportError:
         dsk4 = dsk2
     dsk5 = fuse_getitem(dsk4, dataframe_from_ctable, 3)
-    dsk6, _ = cull(dsk5, keys)
-    return dsk6
+    if read_parquet_row_group:
+        dsk6 = fuse_getitem(dsk5, read_parquet_row_group, 4)
```

Comment:
I would appreciate explaining how the fuse works: I assume df[cols] becomes read_row_group(..., cols,...)